### PR TITLE
ZEPPELIN-337 Configurable ContextPath

### PR DESCRIPTION
**Pull request by echarles**
Allow to configure the context path of the webapp via zeppelin.server.context.path property or ZEPPELIN_CONTEXT_PATH env variable + add doc on this

### CODE REVIEW

**Comment by Leemoonsoo** [comment]
Thanks for the nice feature. Could you rebase and trigger CI?

**Comment by echarles** [comment]
From https://s3.amazonaws.com/archive.travis-ci.org/jobs/88620950/log.txt, the build is success

[INFO] Zeppelin: Packaging distribution ................... SUCCESS [ 35.204 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:26 min

Any reason for the red flag on github?

Should I rebase and push again, or close/reopen to trigger the CI?

**Comment by Leemoonsoo** [comment]
it's because of 
```
[0K$ ./testing/startSparkCluster.sh 1.4.0 2.3
SPARK_HOME is /home/travis/build/apache/incubator-zeppelin/spark-1.4.0-bin-hadoop2.3 


No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.
```
issue resolved by #380. This time, you'll need rebase and push to apply #380.

File: docs/docs/install/install.md [hunk 1]
```diff
@@ -70,6 +70,12 @@ Configuration can be done by both environment variable(conf/zeppelin-env.sh) and
     <td>Zeppelin server port. Note that port+1 is used for web socket</td>
   </tr>
   <tr>
+    <td>ZEPPELIN_CONTEXT_PATH</td>
```
Comment (felixcheung) [inline docs/docs/install/install.md]: could this be `ZEPPELIN_SERVER_CONTEXT_PATH`?
there is a object called ZeppelinContext in the Spark interpreter group and this name might be confusing

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java [hunk 2]
```diff
@@ -268,6 +268,10 @@ public int getServerPort() {
     return getInt(ConfVars.ZEPPELIN_PORT);
   }
 
+  public String getContextPath() {
```
Comment (felixcheung) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java]: same for naming of this..

Comment (echarles) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java]: @felixcheung Makes sense. Just pushed an update that does this.

**Comment by echarles** [comment]
Any more feedback on this?

**Comment by Leemoonsoo** [comment]
I'm not sure why CI is failing. You can trigger CI again.
And the branch seems removed, can not try anymore.

**Comment by echarles** [comment]
Please use #429 to track this.

**Pull request closed without merging**